src/Mod/Sketcher/Gui/DrawSketchHandlerRotate.h

```diff
@@ -0,0 +1,632 @@
+                    newConstr->Type = cstr->Type;
+                    newConstr->FirstPos = cstr->FirstPos;
+                    newConstr->SecondPos = cstr->SecondPos;
+                    newConstr->ThirdPos = cstr->ThirdPos;
+                    newConstr->First = firstIndexi;
+
+                    if ((cstr->Type == Symmetric || cstr->Type == Tangent
+                         || cstr->Type == Perpendicular)
+                        && firstIndex >= 0 && secondIndex >= 0 && thirdIndex >= 0) {
+                        newConstr->Second = secondIndexi;
+                        newConstr->Third = thirdIndexi;
+                    }
+                    else if ((cstr->Type == Coincident || cstr->Type == Tangent
+                              || cstr->Type == Symmetric || cstr->Type == Perpendicular
+                              || cstr->Type == Parallel || cstr->Type == Equal
+                              || cstr->Type == PointOnObject)
+                             && firstIndex >= 0 && secondIndex >= 0
+                             && thirdIndex == GeoEnum::GeoUndef) {
+                        newConstr->Second = secondIndexi;
+                    }
+                    else if ((cstr->Type == Radius || cstr->Type == Diameter
+                              || cstr->Type == Weight)
+                             && firstIndex >= 0) {
+                        if (deleteOriginal || !cloneConstraints) {
+                            newConstr->setValue(cstr->getValue());
+                        }
+                        else {
+                            newConstr->Type = Equal;
+                            newConstr->First = cstr->First;
+                            newConstr->Second = firstIndexi;
+                        }
+                    }
+                    else if ((cstr->Type == Distance || cstr->Type == DistanceX
+                              || cstr->Type == DistanceY)
+                             && firstIndex >= 0 && secondIndex >= 0) {
+                        if (!deleteOriginal && cloneConstraints
+                            && cstr->First == cstr->Second) {  // only line distances
+                            if (indexInVec(geoIdsWhoAlreadyHasEqual, secondIndexi) != -1) {
+                                continue;
+                            }
+                            newConstr->Type = Equal;
+                            newConstr->First = cstr->First;
+                            newConstr->Second = secondIndexi;
+                            geoIdsWhoAlreadyHasEqual.push_back(secondIndexi);
+                        }
+                        else {
+                            newConstr->Second = secondIndexi;
+                        }
+                    }
+                    else if ((cstr->Type == Block) && firstIndex >= 0) {
+                        newConstr->First = firstIndexi;
+                    }
+                    else {
+                        continue;

```

Comment:
It's actually quite complex to handle as you describe. I was trying to do it, and it's becoming too complex.
Because there are actually many special cases where we don't want to copy the constraint. For example perpendicular between a line and an axis. Which is why there are test `&& firstIndex >= 0 && secondIndex >= 0`

So testing before will duplicate many things.

Reply:
I realised that later on. That is why I did not pursue it and move on. I am satisfied that we understand that we do this by way of exception.